iscript/tests/test_mac.py

```diff
@@ -439,25 +439,34 @@ def test_get_bundle_id(mocker, counter):
 
 # get_uuid_from_log {{{1
 @pytest.mark.parametrize(
-    "uuid, raises",
+    "uuid, raises, extra",
     (
-        ("07307e2c-db26-494c-8630-cfa239d4b86b", False),
-        ("d4d31c49-c075-4ea1-bb7f-150c74f608e1", False),
-        ("d4d31c49-c075-4ea1-bb7f-150c74f608e1", "missing file"),
-        ("%%%%\\\\=", "missing uuid"),
+        ("07307e2c-db26-494c-8630-cfa239d4b86b", False, ""),
+        ("d4d31c49-c075-4ea1-bb7f-150c74f608e1", False, "Blah blah blah\nFoo bar baz"),
+        ("d4d31c49-c075-4ea1-bb7f-150c74f608e1", "missing file", ""),
+        ("%%%%\\\\=", "missing uuid", ""),
+        (
+            "07307e2c-db26-494c-8630-cfa239d4b86b",
+            ThrottledNotarization,
+            'altool[15766:50391190] *** Error: ERROR ITMS-10004: "You have reached your upload limit of 20 software packages per minute. Pause your uploads, then reduce the number of software packages you upload per minute.',
+        ),
+        ("d4d31c49-c075-4ea1-bb7f-150c74f608e1", UnknownNotarizationError, "What the! It looks like you've hit an ERROR of some sort"),
     ),
 )
-def test_get_uuid_from_log(tmpdir, uuid, raises):
+def test_get_uuid_from_log(tmpdir, uuid, raises, extra):
     """``get_uuid_from_log`` returns the correct uuid from the logfile if present.
     It raises if it has problems finding the uuid in the log.
 
     """
     log_path = os.path.join(str(tmpdir), "log")
     if raises != "missing file":
         with open(log_path, "w") as fh:
-            fh.write("foo\nbar\nbaz\n RequestUUID = {} \nblah\n".format(uuid))
+            fh.write(f"foo\nbar\nbaz\n RequestUUID = {uuid} \n{extra}\nblah\n")
```

Comment:
Nit, this could be closer to reality if the output written depended on the exception we expect to raise (aka no RequestUUID would be present when throttled).

Reply:
Hm. I think we do get a UUID when throttled, which is part of the problem. (If we didn't get a RequestUUID when throttled, an unpatched `get_uuid_from_log` wouldn't find a UUID, and we would raise an `IScriptError` during notarization-part-1. For some reason they are sending a RequestUUID and `altool` exits 0. The only sign that there was an issue is the `ERROR ITMS-10004`. We're only realizing that the UUID was throttled in the poller, many hours later.)

Reply:
Hm. We did log output with the `ERROR ITMS-10004` but no `RequestUUID`, though, didn't we? Now I'm confused where we get the UUID from.

Reply:
Aha! `2020-04-10 02:19:54,437 - scriptworker_client.utils - INFO - [2020-04-10 02:19:53 GMT] <main> DBG-X:   parameter RequestUUID = 60707656-fd64-430b-ae00-1c4c8fc69463` From https://firefoxci.taskcluster-artifacts.net/VDm1wKy3TK-6VkzvNsqa3Q/0/public/logs/live_backing.log

Reply:
Ah, that makes sense now. Lets disregard my comment.